src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -810,31 +811,62 @@ private LanguageExpression ConvertBinary(BinaryOperationSyntax syntax)
 
         private LanguageExpression ConvertUnary(UnaryOperationSyntax syntax)
         {
-            LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
-
             switch (syntax.Operator)
             {
                 case UnaryOperator.Not:
+                    LanguageExpression convertedOperand = ConvertExpression(syntax.Expression);
                     return CreateFunction("not", convertedOperand);
 
                 case UnaryOperator.Minus:
-                    if (convertedOperand is JTokenExpression literal && literal.Value.Type == JTokenType.Integer)
+                    if (syntax.Expression is IntegerLiteralSyntax integerLiteral)
                     {
-                        // invert the integer literal
-                        int literalValue = literal.Value.Value<int>();
-                        return new JTokenExpression(-literalValue);
+                        //do codegen logic
```

Comment:
It would be good to write a brief comment here explaining why you're shortcutting the integer parsing logic, so that whoever is next looking at this piece of code understands the context and doesn't try to refactor it.